### PR TITLE
Add propTypes in the README

### PR DESCRIPTION
**Pull request by nicolobruno**
## Description

- Add proTypes in the components in the README.
- Add description of TextArea component in the README.
- Corrections lint.

## New aditions

* Changed props list to table
* Added **Type**, **Required** and **Description** info
* Changed `js` to `jsx` syntax highlighting

### Screenshot
**Example**
image

**Base Styles**
captura de pantalla 2018-10-01 a la s 16 13 44

### CODE REVIEW

File: generators/app/templates/src/app/components/TextArea/index.js [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-function TextArea({ onChange, onBlur, onFocus, value }) {
+function TextArea({ className, onChange, onBlur, onFocus, value }) {
   return (
-    <textarea style={styles.textarea} onChange={onChange} onBlur={onBlur} onFocus={onFocus} value={value} />
+    <textarea className={className} onChange={onChange} onBlur={onBlur} onFocus={onFocus} value={value} />
   );
 }
 
 TextArea.propTypes = {
+  className: PropTypes.string,
```
Comment (FrankIglesias) [inline generators/app/templates/src/app/components/TextArea/index.js:11]: add a default prop for this like ''

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

**Comment by nicolobruno** [comment]
I had to add `"eslint-config-react-app": "^2.1.0" ` for run lint. This updated the package-lock.json. Please we check this is ok @FrankIglesias  @damfinkel

**Review by FrankIglesias — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -138,48 +187,56 @@ export default withSpinner({
...
+| | middle | ``` justify-content: center ```|
+| | bottom | ``` justify-content: flex-end ```|
+| | space-between | ``` justify-content: space-between ```|
+| | space-around | ``` justify-content: space-around ```|
+| | wrap | ``` flex-wrap: wrap ```|
+| | reverse | ``` flex-direction: row-reverse ```|
+| | self-top | ``` align-self: flex-start ```|
+| | self-middle | ``` align-self: center ```|
+| | self-bottom | ``` align-self: flex-end ```|
+| | self-stretch | ``` alingn-self: self-stretch ```|
```
Comment (damfinkel) [inline README.md]: align-self

File: README.md [hunk 3]
```diff
@@ -138,48 +187,56 @@ export default withSpinner({
...
+| | half-width | ``` width: 50% ``` |
+| | quarter-width | ``` width: 25% ```|
+| | | |
+| margins.scss | | scale: 10px |
+| | m-top-x | ``` margin-top: {x * scale}px ```|
+| | m-right-x | ``` margin-right: {x * scale}px ```|
+| | m-bottom-x | ``` margin-bottom: {x * scale}px ```|
+| | m-left-x | ``` margin-left: {x * scale}px ```|
+| | | |
+| layout.scss| row | ``` display: flex flex-direction: row ```|
```
Comment (damfinkel) [inline README.md]: Some of these properties change depending on which flex-direction they have. For example:
`row middle` is `align-items: center;` and
`column middle` is `justify-content: center;`

Maybe we could have one row per each of these combinations for those classes that change

File: README.md [hunk 4]
```diff
@@ -138,48 +187,56 @@ export default withSpinner({
...
     type="text"
     className={`row center m-top-2 ${styles.input}`}
   />
 </div>
 ```
 
-"row center middle full-width" is equals a this:
+```jsx
+className="row center middle full-width"
+// Is equals a this:
```
Comment (damfinkel) [inline README.md]: Is equal to this:

File: generators/app/templates/src/app/components/Spinner/index.js [hunk 5]
```diff
@@ -11,7 +11,7 @@ export function withSpinner({
   typeLoading,
   colorSpinner
 }) {
-  function Spinner({ loading }) {
+  function Spinner({ loading, props }) {
```
Comment (damfinkel) [inline generators/app/templates/src/app/components/Spinner/index.js]: `function Spinner({ loading, ...props }) {`

**Review by damfinkel — CHANGES_REQUESTED** [review]

**Review by damfinkel — APPROVED** [review]

**Pull request merged**